mountpoint-s3/src/prefetch/part_stream.rs

```diff
@@ -223,12 +227,35 @@ where
                                 let checksum_bytes = ChecksummedBytes::new(chunk);
                                 let part = Part::new(id.clone(), curr_offset, checksum_bytes);
                                 curr_offset += part.len() as u64;
-                                part_queue_producer.push(Ok(part));
+                                part_queue_producer.push(Ok(part), Some(next_read_window_offset));
+                            }
+                            debug_assert!(
+                                curr_offset <= last_offset,
+                                "current offset must not be larger than last offset"
+                            );
+
+                            if remaining_window_size > 0 {
```

Comment:
It's not clear to me why we are calling `increment_read_window` here or why we need the sender/receiver pair. I'd argue this logic requires some comment or maybe a rethink.